syncer/object.go

```diff
@@ -30,11 +31,44 @@ type ObjectSyncer struct {
 	previousObject runtime.Object
 }
 
+// objectWithoutSecretData returns the object without secretData
+func objectWithoutSecretData(obj runtime.Object) runtime.Object {
```

Comment:
`objectWithoutSecretData` implies more creating an object without secret data.
I would rather call it `stripSecrets` or something similar.